pydocx/export/html.py

```diff
@@ -7,24 +7,112 @@
 )
 
 import base64
-import xml.sax.saxutils
+import posixpath
+from itertools import chain
+from collections import defaultdict
 
 from pydocx.constants import (
+    INDENTATION_FIRST_LINE,
+    INDENTATION_LEFT,
+    INDENTATION_RIGHT,
+    JUSTIFY_CENTER,
+    JUSTIFY_LEFT,
+    JUSTIFY_RIGHT,
     POINTS_PER_EM,
     PYDOCX_STYLES,
     TWIPS_PER_POINT,
+    EMUS_PER_PIXEL,
 )
 from pydocx.export.base import PyDocXExporter
+from pydocx.openxml import wordprocessing
+from pydocx.util.uri import uri_is_external
 from pydocx.util.xml import (
     convert_dictionary_to_html_attributes,
     convert_dictionary_to_style_fragment,
 )
 
 
-class PyDocXHTMLExporter(PyDocXExporter):
+def convert_twips_to_ems(value):
+    '''
+    >>> convert_twips_to_ems(30)
+    0.125
+    '''
+    return value / TWIPS_PER_POINT / POINTS_PER_EM
+
+
+def convert_emus_to_pixels(emus):
+    return emus / EMUS_PER_PIXEL
+
+
+def get_first_from_sequence(sequence, default=None):
+    first_result = default
+    try:
+        first_result = next(sequence)
+    except StopIteration:
+        pass
+    return first_result
+
+
+class HtmlTag(object):
```

Comment:
I really like this extraction. It forces us to make sure all tags are created the same way (pesky dictionary based attrs) and deals with self closing tags by default.

+1